analysis/model/make_model_input_gi_bleeds.R

```diff
@@ -0,0 +1,144 @@
+                           "cov_cat_region",
+                           "cov_cat_deprivation",
+                           "cov_cat_smoking_status",
+                           "cov_bin_carehome_status",
+                           "cov_bin_obesity",
+                           "cov_bin_healthcare_worker",
+                           "cov_bin_alcohol_above_limits",
+                           "cov_bin_cholelisthiasis",
+                           "cov_bin_nsaid_bnf",
+                           "cov_bin_aspirin_bnf",
+                           "cov_bin_h_pylori_infection",
+                           "cov_bin_anticoagulants_bnf",
+                           "cov_bin_antidepressants_bnf",
+                           "cov_bin_gi_operations",                                                            
+                           "sub_cat_covid19_hospital",
+                           "sub_bin_covid19_confirmed_history"))]
+  
+  print (paste0("nrow after restrict to required variables : ",nrow(input)))
+  print(summary(input$cov_num_age))
+  
+  input <- dplyr::rename(input, 
+                         "out_date" =active_analyses$outcome[i],
+                         "exp_date" = active_analyses$exposure[i])
+  print (paste0("nrow after rename : ",nrow(input)))
+  
+  #End_Date outcome and End_Date exposure 
+  
+  input <- input %>% 
+           mutate(end_date_outcome = as.Date(ifelse(!is.na(out_date), out_date, study_end_date)))
+                      
+  input <- input %>% 
+    mutate(end_date_exposure = as.Date(ifelse(!is.na(exp_date), exp_date, study_end_date)))
+                                   
+  input <- input %>% 
+    dplyr::mutate(out_date = replace(out_date, which(out_date>end_date_outcome | out_date<index_date), NA),
+                  exp_date =  replace(exp_date, which(exp_date>end_date_exposure | exp_date<index_date), NA),
+                  sub_cat_covid19_hospital = replace(sub_cat_covid19_hospital, which(is.na(exp_date)),"no_infection"))
+  
+  print (paste0("nrow after replace : ",nrow(input)))
+  
+  # Update end date to be outcome date where applicable ------------------------
+  print('Update end date to be outcome date where applicable')
+  
+  input <- input %>% 
+    dplyr::rowwise() %>% 
+    dplyr::mutate(end_date_outcome = min(end_date_outcome, out_date, na.rm = TRUE))
+  
+  print (paste0("nrow after Update end date to be outcome date : ",nrow(input)))
+  
+  
+  # Make model input: main -------------------------------------------------------
+  
+  if (active_analyses$analysis[i]=="main") {
+    
+    print('Make model input: main')
+    df <- input[input$sub_bin_covid19_confirmed_history==FALSE,]
+    df[,colnames(df)[grepl("sub_",colnames(df))]] <- NULL
+    check_vitals(df)
+    readr::write_rds(df, file.path("output", paste0("model_input-",active_analyses$name[i],".rds")),compress="gz")
+    print(paste0("Saved: output/model_input-",active_analyses$name[i],".rds"))
```

Comment:
you either need to change the file name here or change the name in active analyses (by adding _gi_bleeds) 
(the latter is better)

Reply:
I can't see the requested change